diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter4/NoVarRuleWarnTest.kt

```diff
@@ -0,0 +1,87 @@
+package org.cqfn.diktat.ruleset.chapter4
+
+import com.pinterest.ktlint.core.LintError
+import generated.WarningNames.SAY_NO_TO_VAR
+import org.cqfn.diktat.ruleset.constants.Warnings
+import org.cqfn.diktat.ruleset.rules.DIKTAT_RULE_SET_ID
+import org.cqfn.diktat.ruleset.rules.ImmutableValNoVarRule
+import org.cqfn.diktat.util.LintTestBase
+import org.junit.jupiter.api.Tag
+import org.junit.jupiter.api.Test
+
+class NoVarRuleWarnTest : LintTestBase(::ImmutableValNoVarRule) {
+
+    private val ruleId = "$DIKTAT_RULE_SET_ID:no-var-rule"
+
+    @Test
+    @Tag(SAY_NO_TO_VAR)
+    fun `valid case where x is used in while loop as some counter`() {
+        lintMethod(
+                """
+                    | fun foo() { 
+                    |     var x = 0
+                    |     while (x < 10) {
+                    |        x++
+                    |     }
+                    | }
+                """.trimMargin()
+        )
+    }
+
+    @Test
+    @Tag(SAY_NO_TO_VAR)
+    fun `valid case where y is used in for each loop as some counter, but a is not`() {
+        lintMethod(
+                """
+                    | fun foo() { 
+                    |     var a = emptyList()
+                    |     var y = 0
+                    |     a.forEach { x ->
+                    |        y = x + 1
+                    |     }
+                    | }
+                """.trimMargin(),
+                LintError(2, 6, ruleId, "${Warnings.SAY_NO_TO_VAR.warnText()} var a = emptyList()", false)
+        )
+    }
+
+    @Test
+    @Tag(SAY_NO_TO_VAR)
+    fun `For loop with internal counter`() {
+        lintMethod(
+                """
+                    | fun foo() { 
+                    |     for (x in 0..10) println(x)
+                    | }
+                """.trimMargin()
+        )
+    }
+
+    @Test
+    @Tag(SAY_NO_TO_VAR)
+    fun `var in class`() {
+        lintMethod(
+                """
+                    | class A { 
+                    |     var a = 0
+                    | }
+                """.trimMargin()
+        )
+    }
+
+    @Test
+    @Tag(SAY_NO_TO_VAR)
+    fun `var used simply in function`() {
```

Comment:
Do you suggest in this case for user to create a chain of calls? Are you sure it's reasonable to forbid using `var`s for all cases with reassignment?

Reply:
we can suggest, they can void it
but definitely if they will be using something like:
```
 val a = 0
 val aPLus15 = a + 15
```

or directly `return a + 15 + 56 `

this is much more threadsafe and compact

Reply:
I mean, it makes sense for simple binary expressions, but if they have
```kotlin
var a = 0
println(a)
a = a + 15
return a
```
do you want them to replace it with
```kotlin
val a = 0.also {
   println(it)
}
.plus(15)
return a
```
?

Reply:
Yeah, got it, you are right. 
I need to do more accurate check after I will create a proper mechanism of detecting assignments